### ISSUE TITLE
naming of spill/shuffle/merge components

### ISSUE THREAD

**Issue by mccanne** [brimsec/zq#1446]
Now that we have experiences piecing together the components to do spills, merged, and shuffles, we should reconsider re-naming things to embrace holistic naming architecure.  e.g., oderedmergeproc is really doing a join-key merge, etc.

There is work here to define all the components from the name of orderedmergeproc to the name of the MergeOrderField field in parallel proc to the name of parallel proc itself, then to propose some options for an overall naming scheme.

**Comment by mccanne** [comment]
This feels like low priority on the one hand.  On the other, doing this work will help us clarify architectural and design decision for surrounding componentry.  Choosing the right words is important in framing the problems and finding the solutions.
